src/crc-setup.ts

```diff
@@ -0,0 +1,108 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as extensionApi from '@podman-desktop/api';
+import { execPromise, getCrcCli } from './crc-cli';
+import type { Preset } from './daemon-commander';
+import { productName } from './util';
+
+interface PresetQuickPickItem extends extensionApi.QuickPickItem {
+  data: Preset;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export async function setUpCrc(logger: extensionApi.Logger, askForPreset = false): Promise<boolean> {
+  if (askForPreset) {
+    const preset = await extensionApi.window.showQuickPick<PresetQuickPickItem>(createPresetItems(), {
+      canPickMany: false,
+      title: `Select ${productName} Preset`,
+      placeHolder: `Select ${productName} Preset`,
+    });
+    if (!preset) {
+      extensionApi.window.showNotification({
+        title: productName,
+        body: 'Default preset will used.',
+      });
+    } else {
+      await execPromise(getCrcCli(), ['config', 'set', 'preset', preset.data]);
+    }
+  }
+
+  const setupBar = extensionApi.window.createStatusBarItem('RIGHT', 2000);
+  try {
+    setupBar.text = `Configuring ${productName}...`;
+    setupBar.show();
+    await execPromise(getCrcCli(), ['setup'], {
+      logger: {
+        error: (data: string) => {
```

Comment:
do we use stderr? ouch...

Reply:
Yes

Reply:
iirc, this was for incorporating `-o json`, when you want a json o/p we don't want log messages to appear, so logs are sent to stderr and json o/p to stdout